syntax = "proto3";
package game;

message Game {
string id = 1;
string title = 2;
string description = 3;
string type = 4;
string prix = 5;
}

message GetGameRequest {
string game_id = 1;
}

message GetGameResponse {
Game game = 1;
}

message SearchGamesRequest {
string query = 1;
}

message SearchGamesResponse {
repeated Game games = 1;
}

message AddGameRequest {
    string title = 1;
    string description = 2;
    string type = 3;
    string prix = 4;
}

message AddGameResponse {
    Game game = 1;
}

service GameService {
rpc GetGame(GetGameRequest) returns (GetGameResponse);
rpc SearchGames(SearchGamesRequest) returns (SearchGamesResponse);
rpc AddGame(AddGameRequest) returns (AddGameResponse);

}